### PR TITLE
Enable navbar for export defaults

### LINKED ISSUE #2202: Support "export default class/function" in the navbar.
The new es6 import/export work is going in.  As part of it, it filters out classes/functions from the navbar  if they have no name.  But we'll want to show these guys (possibly has "default class", or some other term) in the navbar.

### PR DESCRIPTION
**Pull request by mhegazy**
Fixes #2202

### CODE REVIEW

File: src/services/navigationBar.ts [hunk 1]
```diff
@@ -424,10 +424,10 @@ module ts.NavigationBar {
             }
 
             function createFunctionItem(node: FunctionDeclaration) {
-                if (node.name && node.body && node.body.kind === SyntaxKind.Block) {
+                if ((node.name || node.flags & NodeFlags.Default) && node.body && node.body.kind === SyntaxKind.Block) {
                     var childItems = getItemsWorker(sortNodes((<Block>node.body).statements), createChildItem);
 
-                    return getNavigationBarItem(node.name.text,
+                    return getNavigationBarItem((node.flags & NodeFlags.Default) ? "default": node.name.text ,
```
Comment (DanielRosenwasser) [inline src/services/navigationBar.ts]: Stray space; also I think this was a little cleaner when you just used an intermediate variable.

Comment (sheetalkamat) [inline src/services/navigationBar.ts]: If the syntax is of type export default function foo() {} shouldn't you be showing foo instead of default?

Comment (mhegazy) [inline src/services/navigationBar.ts]: done.

**Comment by mhegazy** [comment]
moved to #2368

**Pull request closed without merging**